src/Repositories/Behaviors/HandleBlocks.php

```diff
@@ -174,25 +176,38 @@ private function updateBlock(
         return $blockCreated;
     }
 
-    private function validate(array $formData, int $id, array $basicRules, array $translatedFieldRules): void
+    private function validate(array $formData, int $id, array $basicRules, array $translatedFieldRules, array $messages): void
     {
         $finalValidator = $this->blockValidator;
         foreach ($translatedFieldRules as $field => $rules) {
             foreach (config('translatable.locales') as $locale) {
-                $data = $formData[$field][$locale] ?? null;
-                $validator = Validator::make([$field => $data], [$field => $rules]);
+                $validator = Validator::make($formData, ["$field.$locale" => $rules], $messages);
                 foreach ($validator->messages()->getMessages() as $key => $errors) {
                     foreach ($errors as $error) {
+                        if ($this->errorMessageIsOnNestedBlock($key)) {
+                            $blockInfo = $this->makeValidationInfoForNestedBlocks($id, $key, $formData, true);
+
+                            $id = $blockInfo['nestedBlockId'];
+                            $key = $blockInfo['nestedField'];
+                        }
+
                         $finalValidator->getMessageBag()->add("blocks.$id" . "[$key][$locale]", $error);
                         $finalValidator->getMessageBag()->add("blocks.$locale", 'Failed');
                     }
                 }
             }
         }
         foreach ($basicRules as $field => $rules) {
-            $validator = Validator::make([$field => $formData[$field] ?? null], [$field => $rules]);
+            $validator = Validator::make($formData, [$field => $rules], $messages);
```

Comment:
Couldn't we just drop the `foreach` loop at this point and pass `$basicRules` directly to the validator?

I just ran into the same issue validating fields and media browsers within a repeater of a block, thought I couldn't be the first and found this PR, so thank you for that 😄